src/NHibernate.Test/NHSpecificTest/GH1496/AuditEventListener.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NHibernate.Event;
+
+namespace NHibernate.Test.NHSpecificTest.GH1496
+{
+	public class AuditEventListener : IPostUpdateEventListener
+	{
+		public IList<Item> ModifiedItems { get; set; } = new List<Item>();
+		private bool isActive = false;
+
+		public void Start()
+		{
+			ModifiedItems.Clear();
+			isActive = true;
+		}
+
+		public void Stop()
+		{
+			isActive = false;
+		}
+
+		public async Task OnPostUpdateAsync(PostUpdateEvent @event, CancellationToken cancellationToken)
```

Comment:
Normally there is a no need to write this async method too. The async generator will do it for you too. Of course this means that at first your class will lack some interface method implementations and the project will not compile till you run the async generator, but this is the preferred way.